apps/els_lsp/src/els_completion_provider.erl

```diff
@@ -44,8 +44,18 @@ handle_request({completion, Params}, State) ->
                             }
    , <<"textDocument">> := #{<<"uri">> := Uri}
    } = Params,
-  ok = els_index_buffer:flush(Uri),
```

Comment:
A bit of context may help here.

Completion requests typically come at the same time as edits. With the current implementation of the `els_index_buffer` it could happen that, during a completion, edits are still not processed (there's some basic protection against this, but things can still fail).

The completion provider assumes a specific length in the line when reading the text after the flush, but it's not guaranteed that all edits are applied. Fast typers can see the provider (and eventually the whole server) crashing.

With the new approach, the idea is to block callers of the flush for a certain amount of time (right now 200ms), but if edits keep coming the count restarts.

This should also prevent indexing happening multiple times.